clio-client/src/test/scala/org/broadinstitute/clio/client/dispatch/AddArraysSpec.scala

```diff
@@ -0,0 +1,45 @@
+package org.broadinstitute.clio.client.dispatch
+
+import akka.http.scaladsl.model.headers.OAuth2BearerToken
+import org.broadinstitute.clio.client.BaseClientSpec
+import org.broadinstitute.clio.client.commands.AddArrays
+
+class AddArraysSpec extends BaseClientSpec {
+  behavior of "AddArrays"
+
+  implicit val bearerToken: OAuth2BearerToken = testBearer
+
+  val dispatcher = succeedingDispatcher()
+
+  it should "fail if the metadata is not valid json" in {
+    recoverToSucceededIf[IllegalArgumentException] {
+      dispatcher.dispatch(
+        AddArrays(
+          metadataLocation = badMetadataFileLocation,
+          key = testArraysTransferV1Key
+        )
+      )
+    }
+  }
+
+  it should "fail if the json is valid but we can't decode it" in {
+    recoverToSucceededIf[IllegalArgumentException] {
+      dispatcher.dispatch(
+        AddArrays(
+          metadataLocation = metadataPlusExtraFieldsFileLocation,
+          key = testArraysTransferV1Key
+        )
+      )
+    }
+  }
+
+  /**
+    * FIXME: Bogus because server does not support Arrays yet.
```

Comment:
Agreed the tests are bogus, I don't think it's because of anything on the server side. The goal for these tests was to mock away the server to isolate the client. We "succeeded" by mocking away basically everything of value.

Better tests would still run & pass regardless of the state of the server code, but would be more clear / surgical in their mocking.

Reply:
The "Bogus" remark was that these tests are more bypassing now than passing.